src/main/java/com/byznass/tiolktrack/jaxrs/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,83 @@
+package com.byznass.tiolktrack.jaxrs.filter;
+
+import com.byznass.tiolktrack.kernel.handler.AuthenticationHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.ext.Provider;
+
+import static com.byznass.tiolktrack.jaxrs.filter.AuthenticationException.Reason.INVALID_AUTHENTICATION_METHOD;
+import static com.byznass.tiolktrack.jaxrs.filter.AuthenticationException.Reason.INVALID_FORMAT;
+import static javax.ws.rs.core.HttpHeaders.AUTHORIZATION;
+
+@Provider
+@Priority(Priorities.AUTHENTICATION)
+public class AuthenticationFilter implements ContainerRequestFilter {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(AuthenticationFilter.class);
+
+	private static final String AUTHENTICATION_SCHEME = "bearer ";
+	private static final String ALL_LEADING_SPACES = "^ *";
+	private static final String EMPTY = "";
+	private static final String SPACE = " ";
+
+	private final AuthenticationHandler authenticationHandler;
+
+	@Inject
+	public AuthenticationFilter(AuthenticationHandler authenticationHandler) {
+
+		this.authenticationHandler = authenticationHandler;
+	}
+
+	@Override
+	public void filter(ContainerRequestContext requestContext) throws AuthenticationException {
+
+		LOGGER.info("Starting request authentication");
+
+		String authorizationHeader = requestContext.getHeaderString(AUTHORIZATION);
+		validateAuthenticationMethod(authorizationHeader);
+
+		String userIdWithToken = removeAuthMethod(authorizationHeader);
+		validateUserToken(userIdWithToken);
+	}
+
+	private void validateAuthenticationMethod(String authorizationHeader) throws AuthenticationException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.byznass.tiolktrack.jaxrs.filter.AuthenticationException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.tiolktrack.byznass.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)